### ISSUE TITLE
Add check for `implies` argument type

### ISSUE THREAD

**Issue by canopy-js-user** [tj/commander.js#1852]
Hi there, twice I've accidentally written a line of code like this:

```
.addOption(new Option(
    '--port <number>', 
    'Which port to run the server on for sync mode'
).default(undefined).implies('sync'))
```

Ie, I want the `--port` option to imply the boolean flag `sync`. However, `sync` is expecting an object of implied key-value pairs. However, as is, this fails silently and converts the string/array `'sync'` into an object, and merges it with the rest of the options:

```
{
  '0': 's',
  '1': 'y',
  '2': 'n',
  '3': 'c',
  editor: true,
  logging: true,
  open: true,
  backup: true,
  sync: true,
  port: '4004'
}
```

I'm proposing a check that the argument to `implies` is an object and not a string, and if it is a string, error. (Or, interpret the string as the name of a boolean option that the programmer is asserting should be true.)

Thanks!

**Comment by canopy-js-user** [comment]
(Another reason I think this an easy mistake is that many of the option methods _do_ take strings, eg `.conflics`. Thanks!)

**Comment by shadowspawn** [comment]
Commander does not have a lot of runtime checks on the parameters, but I am happy to consider adding checks or improving behaviour for routines that are easy to get wrong.

(A past one that was added was #1655.)

**Comment by canopy-js-user** [comment]
Makes sense, thanks!

**Comment by shadowspawn** [comment]
For interest, we added support in `.conflicts()` for a string or an array after making a similar mistake during development: https://github.com/tj/commander.js/pull/1678#issuecomment-1047210739

**Comment by shadowspawn** [comment]
Shipped in 10.0.1

Thanks @canopy-js-user

**Comment by canopy-js-user** [comment]
@shadowspawn thank _you_!
